src/python/pants/engine/isolated_process.py

```diff
@@ -23,28 +24,25 @@ class ExecuteProcessRequest(datatype([
   ('argv', tuple),
   ('input_files', DirectoryDigest),
   ('description', text_type),
-  ('env', tuple),
-  ('output_files', tuple),
-  ('output_directories', tuple),
+  # TODO: allow inferring a default value if a type is provided which has a 0-arg constructor.
+  F('env', Exactly(tuple, dict, type(None)), default_value=None),
```

Comment:
Why is this `None`able? I would remove `type(None)` from its type constraints, and default it to `()` or `{}`...

Reply:
This sounds like a correct analysis and I will do that.